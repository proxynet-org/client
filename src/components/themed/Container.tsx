import { View as DefaultView } from 'react-native';

export type ViewProps = DefaultView['props'];

export function Container(props: ViewProps) {
  const { style, ...otherProps } = props;

  return (
    <DefaultView
      style={[
        {
          backgroundColor: 'transparent',
          flex: 1,
          justifyContent: 'center',
          alignItems: 'center',
        },
        style,
      ]}
      {...otherProps}
    />
  );
}
