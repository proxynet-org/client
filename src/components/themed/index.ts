export { View } from './View';
export { FormInput } from './FormInput';
