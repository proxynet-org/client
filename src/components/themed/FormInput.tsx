import { useState } from 'react';
import { Input, InputProps, useTheme } from '@rneui/themed';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

export function FormInput({
  placeholder,
  value,
  errorMessage,
  onChangeText,
  onBlur,
  secureTextEntry,
  leftIconName,
}: InputProps & {
  leftIconName: string;
}) {
  const [isFocused, setIsFocused] = useState(false);

  const { theme } = useTheme();

  return (
    <Input
      secureTextEntry={secureTextEntry}
      value={value}
      errorMessage={errorMessage}
      onChangeText={onChangeText}
      onFocus={() => setIsFocused(true)}
      onBlur={(e) => {
        setIsFocused(false);
        onBlur?.(e);
      }}
      label={placeholder}
      placeholder={placeholder}
      leftIcon={
        <Icon name={leftIconName} size={20} color={theme.colors.black} />
      }
      containerStyle={{
        backgroundColor: isFocused ? theme.colors.searchBg : 'transparent',
        borderRadius: 25,
      }}
      placeholderTextColor="white"
      labelStyle={{
        paddingLeft: 25,
        fontSize: 12,
        color: value ? theme.colors.black : 'transparent',
        fontWeight: 'bold',
      }}
      inputContainerStyle={{
        height: 25,
        borderBottomWidth: 0,
      }}
      errorStyle={{
        paddingLeft: 25,
        margin: 0,
        color: theme.colors.error,
      }}
      inputStyle={{ color: theme.colors.black, fontWeight: 'bold' }}
    />
  );
}

export default FormInput;
