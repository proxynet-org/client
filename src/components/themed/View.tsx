import {
  SafeAreaView as DefaultSafeAreaView,
  View as DefaultView,
} from 'react-native';

import Animated from 'react-native-reanimated';

import { useTheme } from '@rneui/themed';

export type SafeAreaViewProps = DefaultSafeAreaView['props'];

export function SafeAreaView(props: SafeAreaViewProps) {
  const { style, ...otherProps } = props;
  const { theme } = useTheme();

  return (
    <DefaultSafeAreaView
      style={[{ backgroundColor: theme.colors.background }, style]}
      {...otherProps}
    />
  );
}

export type ViewProps = DefaultView['props'];

export function View(props: ViewProps) {
  const { style, ...otherProps } = props;
  const { theme } = useTheme();

  return (
    <DefaultView
      style={[{ backgroundColor: theme.colors.background }, style]}
      {...otherProps}
    />
  );
}

export type AnimatedViewProps = Animated.View['props'];

export function AnimatedView(props: AnimatedViewProps) {
  const { style, ...otherProps } = props;
  const { theme } = useTheme();

  return (
    <Animated.View
      style={[{ backgroundColor: theme.colors.background }, style]}
      {...otherProps}
    />
  );
}
