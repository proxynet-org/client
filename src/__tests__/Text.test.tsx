import * as React from 'react';
import renderer from 'react-test-renderer';

import { Text } from '@rneui/themed';

it(`renders correctly`, () => {
  const tree = renderer.create(<Text>Snapshot test!</Text>).toJSON();

  expect(tree).toMatchSnapshot();
});
