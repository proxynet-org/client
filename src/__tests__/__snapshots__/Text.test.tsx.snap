// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessibilityRole="text"
  style={
    Object {
      "color": "#242424",
    }
  }
>
  Snapshot test!
</Text>
`;
